pkg/gce-pd-csi-driver/node.go

```diff
@@ -733,5 +741,24 @@ func (ns *GCENodeServer) GetVolumeLimits() (int64, error) {
 			return volumeLimitSmall, nil
 		}
 	}
+	gen4MachineTypesPrefix := []string{"c4a-", "c4-", "n4-"}
+	for _, gen4Prefix := range gen4MachineTypesPrefix {
+		if strings.HasPrefix(machineType, gen4Prefix) {
+			cpuString := machineType[strings.LastIndex(machineType, "-")+1:]
+			cpus, err := strconv.ParseInt(cpuString, 10, 64)
+			if err != nil {
+				return -1, fmt.Errorf("invalid cpuString %s for machine type: %v", cpuString, machineType)
```

Comment:
Should we return volumeLimitBig instead of -1 here? Because currently the err is not blocking.

Reply:
Since we know it's a gen4 probably should return the limit small to be conservative.